iOS Client/PodcastsController/PodcastsCoordinator.swift

```diff
@@ -43,10 +48,45 @@ class PodcastsCoordinator: Coordinator, StoreSubscriber {
     
     // MARK: - StoreSubscriber
     func newState(state: MBAppState) {
+        
+        switch state.podcastsState.player {
+        case .initialized:
+            break
+        case .playing:
+            if let podcast = state.podcastsState.selectedPodcast, podcast.guid != currentPlayingPodcast?.guid {
+                start(podcast)
+                currentPlayingPodcast = state.podcastsState.selectedPodcast
+            } else if case .paused = playerState {
+                player.play()
+            }
+        case .paused:
+            player.pause()
+        case .error:
+            break
+        }
+        
+        playerState = state.podcastsState.player
+        
         guard state.navigationState.selectedTab == .podcasts, let newRoute = state.navigationState.routes[.podcasts] else {
             return
         }
         build(newRoute: newRoute)
         route = newRoute
     }
+    
+    func start(_ podcast: MBPodcast) {
```

Comment:
perhaps this function should only accept a string since that's all that it needs

Reply:
having it accept an MBPodcast is partly in violation of the `interface segregation principle`. As a justification, think about how we could reuse this function from a context where we didn't have an MBPodcast

Reply:
of course the classic interface segregation principle